### ISSUE TITLE
SPTPlaybackMetadata: prevTrack is not updated after skipNext()

### ISSUE THREAD

**Issue by brki** [spotify/ios-sdk#832]
## Description
The prevTrack information in the metadata does not reflect the correct state after skipNext() is called.

### Versions
iOS SDK version: beta 25

### Severity
Major

## Steps to Reproduce
1. Call playSpotifyURI() with track "A"
2. In the audioStreaming(_ audioStreaming: SPTAudioStreamingController!, didStartPlayingTrack  trackUri) function, call queueSpotifyURI() with track "B"
3. Wait a few seconds (during which time the metadata is updated, and at which point metadata.nextTrack is set to track "B".
4. Call skipNext()
5. In the audioStreaming(_ audioStreaming: SPTAudioStreamingController!, didStartPlayingTrack  trackUri) function, call queueSpotifyURI() with track "C"
6. Wait a few seconds.  At this point metadata.nextTrack is track "C", and metadata.prevTrack is track "A".

### Actual behaviour
metadata.prevTrack is track "A".

### Expected behaviour
metadata.prevTrack is track "B".

### Additional notes
metadata.prevTrack always stays at track "A", no matter how many tracks are advanced using queueSpotifyURI() and skipNext().  prevTrack should have the trackInformation for the track that was previously playing.

**Comment by brki** [comment]
I noticed in another issue that this is probably the expected behaviour for skipPrevious (cf. https://github.com/spotify/ios-sdk/issues/742#issuecomment-246370891 ).

If so:
* the method should probably be called something like `skipToInitial` instead of `skipPrevious`
* the documentation should be explicit about the behaviour.  As of beta 25, the docstring for `skipPrevious` says `Go to the previous track in the queue`.

It would be good to better document the queue.

I'm just updating my project to use beta-25 from beta-16, and would like to say that while I'm a bit sad to see the queueing functionality has more or less been removed, I'm happy that other things are working better now.  I can definitely understand the argument of preferring to have something simple and stable.  Kudos for the effort!

**Comment by elmateo487** [comment]
This seems to be expected behavior. Tracks are consumed as they are popped off the queue, and skipPrevious is a convenience method to play the first queued song. IMO, it is pretty useless.

**Comment by brki** [comment]
@elmateo487, yes, so it seems.  My hope is that issue will lead to a bit of documentation about the queue (how it works, why you want to use it, when you should or should not call `queueSpotifyURI`, what effect is has on metadata).
